crates/tabby-common/src/api/event.rs

```diff
@@ -95,35 +93,23 @@ pub struct Segments {
 }
 
 pub trait EventLogger: Send + Sync {
-    fn log(&self, e: Event);
-}
-
-#[derive(Serialize, Deserialize)]
-pub struct Log {
-    pub ts: u128,
-    pub event: Event,
-}
-
-pub trait RawEventLogger: Send + Sync {
-    fn log(&self, content: String);
-}
-
-impl<T: RawEventLogger> EventLogger for T {
     fn log(&self, e: Event) {
         let content = serdeconv::to_json_string(&Log {
             ts: timestamp(),
             event: e,
         })
         .unwrap();
 
-        self.log(content);
+        self.log_raw(content);
     }
+
+    fn log_raw(&self, content: String);
```

Comment:
```suggestion
    fn log(&self, content: String);
```

Keep the name

Reply:
Can't do that, there is already a method `log` which takes an `Event` and rust does not allow overloading